### PR TITLE
[telegram] proxy support for telegram binding

### LINKED ISSUE #3527: [telegram] Add SOCKS5 support
<a alt="bounty" href="https://www.bountysource.com/issues/57694823"></a>

Hi! I live in Russia and we have blocked the Telegram ... Could you help with setting up SOCKS5 proxy in Telegars action?

### PR DESCRIPTION
**Pull request by ehorohorin**
This PR adds proxy support for telegram 2.5 binding. 

Fixes #3527

### CODE REVIEW

File: bundles/org.openhab.binding.telegram/README.md [hunk 1]
```diff
@@ -52,6 +52,29 @@ Please note that the things cannot be used to send messages. In order to send a
 | `chatIds`               |         | Yes      | Comma-separated list of chat ids                                                             |
 | `botToken`              |         | Yes      | authentication token                                                                         |
 | `parseMode`             |  None   | No       | Support for formatted messages, values: Markdown or HTML.                                    |
+| `proxyHost`             |  None   | No       | Proxy host for telegram binding, it doesn't affect entire openhab.                           |
```
Comment (ZzetT) [inline bundles/org.openhab.binding.telegram/README.md]: IMHO it's not necessary to tell the user that the host configuration for a THING doesn't affect the OH config.

File: bundles/org.openhab.binding.telegram/src/main/resources/ESH-INF/thing/thing-types.xml [hunk 2]
```diff
@@ -36,6 +36,23 @@
...
+				<label>Proxy Port</label>
+				<description>Enter your proxy port.</description>
+			</parameter>
+			<parameter name="proxyType" type="text" required="false" multiple="false">
+				<label>Proxy Type</label>
+				<options>
+					<option value="SOCKS5">SOCKS5</option>
+					<option value="HTTP">HTTP</option>
+				</options>
+				<default>"SOCKS5"</default>
```
Comment (ZzetT) [inline bundles/org.openhab.binding.telegram/src/main/resources/ESH-INF/thing/thing-types.xml]: In the PaperUI the default value is only shown if it is written without the quotation marks

File: bundles/org.openhab.binding.telegram/README.md [hunk 3]
```diff
@@ -52,6 +52,29 @@ Please note that the things cannot be used to send messages. In order to send a
...
+
+telegram.thing (no proxy):
+```
+Thing telegram:telegramBot:Telegram_Bot [ chatIds="< ID >", botToken="< TOKEN >" ]
+```
+
+
+telegram.thing (SOCKS5 proxy server is used): 
+```
+Thing telegram:telegramBot:Telegram_Bot [ chatIds="< ID >", botToken="< TOKEN >", proxyHost="localhost", proxyPort="9050", proxyType="SOCKS5" ]
```
Comment (ZzetT) [inline bundles/org.openhab.binding.telegram/README.md]: For the id and the token you used some kind of placeholder like "< ID >", but for the proxyHost and the proxyPort you used a specific example and not "something like "< HOST>" and "< PORT >". Can we keep that consistent?

Comment (ehorohorin) [reply bundles/org.openhab.binding.telegram/README.md]: > For the id and the token you used some kind of placeholder like "< ID >", but for the proxyHost and the proxyPort you used a specific example and not "something like "< HOST>" and "< PORT >". Can we keep that consistent?

Sure, I'll doo

File: bundles/org.openhab.binding.telegram/README.md (hunk 1 above)
Comment (ehorohorin) [reply bundles/org.openhab.binding.telegram/README.md]: > IMHO it's not necessary to tell the user that the host configuration for a THING doesn't affect the OH config.

fixed

File: bundles/org.openhab.binding.telegram/README.md (hunk 3 above)
Comment (ehorohorin) [reply bundles/org.openhab.binding.telegram/README.md]: > For the id and the token you used some kind of placeholder like "< ID >", but for the proxyHost and the proxyPort you used a specific example and not "something like "< HOST>" and "< PORT >". Can we keep that consistent?

fixed

**Comment by ehorohorin** [comment]
Is there any way to signoff commits in the past?

**Comment by ZzetT** [comment]
Yes I think that's possible: https://stackoverflow.com/questions/13043357/git-sign-off-previous-commits

**Comment by ZzetT** [comment]
Changes are fine for me. @J-N-K can you review the files as a second reviewer and (if you agree) merge the changes? Thanks!

**Comment by ZzetT** [comment]
Because of https://github.com/openhab/openhab-addons/issues/6702 we have to rebase to 2.5.x and can't commit this to master anymore.

**Comment by ZzetT** [comment]
@ehorohorin Can you do the rebase to the 2.5.x branch? I think only you are able to do such changes. I think it was a bit of bad luck regarding the timing because after you created the branch from master, OH created a new 2.5.x branch which is now the "master" for OH 2.5.x and the actual master will be used for OH3.

**Comment by ehorohorin** [comment]
> @ehorohorin Can you do the rebase to the 2.5.x branch? I think only you are able to do such changes. I think it was a bit of bad luck regarding the timing because after you created the branch from master, OH created a new 2.5.x branch which is now the "master" for OH 2.5.x and the actual master will be used for OH3.

Yeah, will do.
Hope to complete on weekend

**Comment by ehorohorin** [comment]
@ZzetT could you take a look at #6824, please? 

Many thanks

**Comment by J-N-K** [comment]
superseded by #6824

**Pull request closed without merging**